pypsa/plot.py

```diff
@@ -258,11 +260,12 @@ def plot(network, margin=0.05, ax=None, geomap=True, projection=None,
     def as_branch_series(ser):
         # ensure that this function always return a multiindexed series
         if isinstance(ser, dict) and set(ser).issubset(branch_components):
-            return network.branches().loc[branch_components] \
-                          .join(pd.Series(ser, name='value') \
-                          .rename_axis('component')).value
+            return pd.concat(
+                    {c.name: pd.Series(s, index=c.df.index) for c, s in
+                         zip(network.iterate_components(ser.keys()), ser.values())},
+                    names=['component', 'name'])
         elif isinstance(ser, pd.Series) and isinstance(ser.index, pd.MultiIndex):
-                return ser.rename_axis(['component', 'name'])
+                return ser.rename_axis(index=['component', 'name'])
```

Comment:
I'm pretty sure it is
`ser.rename(index=...)`

Reply:
Here is the example which does not work:
```python 
network.branches().rename(index=['component', 'name'])
```
(if this is what you mean) fails as pd.DataFrame.rename wants to have a mapper or a function. 
On the other hand when passing a mapper, as 
```python 
ser = network.branches()
ser.rename(index=dict(zip(ser.index.names, ['component', 'name'])))
```
does not change the index axis names, and would only affect **values** within the index.

Reply:
Seems, I was mistaken (I did mix it up with reindex).

Reply:
Sorry